src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -17,93 +17,125 @@
+            _ownershipEnabled = fields[2].GetBool();
+            _channelPassword = fields[3].GetString();
+            std::string db_BannedList = fields[4].GetString();
+
+            if (!db_BannedList.empty())
             {
-                Field* fields = result->Fetch();
-                _announceEnabled = fields[0].GetBool();
-                _ownershipEnabled = fields[1].GetBool();
-                _channelPassword  = fields[2].GetString();
-                char const* db_BannedList = fields[3].GetCString();
-
-                if (db_BannedList)
+                Tokenizer tokens(db_BannedList, ' ');
+                for (auto const& token : tokens)
                 {
-                    Tokenizer tokens(db_BannedList, ' ');
-                    for (Tokenizer::const_iterator i = tokens.begin(); i != tokens.end(); ++i)
+                    ObjectGuid banned_guid(uint64(atoull(token)));
+                    if (banned_guid)
                     {
-                        ObjectGuid banned_guid(uint64(atoull(*i)));
-                        if (banned_guid)
-                        {
-                            TC_LOG_DEBUG("chat.system", "Channel(%s) loaded bannedStore %s", name.c_str(), banned_guid.ToString().c_str());
-                            _bannedStore.insert(banned_guid);
-                        }
+                        TC_LOG_DEBUG("chat.system", "Channel(%s) loaded player %s into bannedStore", name.c_str(), banned_guid.ToString().c_str());
+                        _bannedStore.insert(banned_guid);
                     }
                 }
             }
-            else // save
-            {
-                stmt = CharacterDatabase.GetPreparedStatement(CHAR_INS_CHANNEL);
-                stmt->setString(0, name);
-                stmt->setUInt32(1, _channelTeam);
-                CharacterDatabase.Execute(stmt);
-                TC_LOG_DEBUG("chat.system", "Channel(%s) saved in database", name.c_str());
-            }
+        }
+        else // save
+        {
+            stmt = CharacterDatabase.GetPreparedStatement(CHAR_INS_CHANNEL);
+            stmt->setString(0, name);
+            stmt->setUInt32(1, _channelTeam);
+            CharacterDatabase.Execute(stmt);
+            TC_LOG_DEBUG("chat.system", "Channel(%s) saved in database", name.c_str());
+        }
+
+        _persistentChannel = true;
+    }
+}
 
-            _persistentChannel = true;
+void Channel::GetChannelName(std::string& channelName, uint32 channelId, LocaleConstant locale, AreaTableEntry const* zoneEntry)
+{
+    if (channelId)
+    {
+        ChatChannelsEntry const* channelEntry = sChatChannelsStore.AssertEntry(channelId);
+        if (!(channelEntry->flags & CHANNEL_DBC_FLAG_GLOBAL))
+        {
+            if (channelEntry->flags & CHANNEL_DBC_FLAG_CITY_ONLY)
+                channelName = Trinity::StringFormat(channelEntry->pattern[locale], sObjectMgr->GetTrinityString(LANG_CHANNEL_CITY, locale));
```

Comment:
What if the server does not have this locale installed and channelEntry->pattern[locale] is empty?